sunpy/util/util.py

```diff
@@ -195,3 +197,35 @@ def hash_file(path):
             sha256.update(data)
 
     return sha256.hexdigest()
+
+
+def get_rectangle_coordinates(bottom_left, *, top_right = None, width: u.deg = None, height: u.deg = None):
+    if width is None and height is None:
+        if not top_right:
+            if bottom_left.shape[0] != 2:
+                raise IndexError("If top_right is not specified bottom_left must have length two or width and height must be provided.")
+            else:
+                top_right = bottom_left[1]
+                bottom_left = bottom_left[0]
+
+        if type(bottom_left) is not type(top_right):
+            raise ValueError("top_right must be of same type as bottom_left")
+
+        if isinstance(bottom_left, (SkyCoord, BaseCoordinateFrame)):
+            top_right = top_right.transform_to(bottom_left.frame.name)
+
+            width = (top_right.spherical.lon - bottom_left.spherical.lon).to(u.deg)   # Getting the difference in Longitudes.
+            height = (top_right.spherical.lat - bottom_left.spherical.lat).to(u.deg)  # Getting the difference in Latitudes.
+
+            if width <= 0*u.deg or height <= 0*u.deg:
+                raise ValueError("top_right must be to the right and above bottom_left.")
+        else:
+            raise ValueError("Invalid value passed for bottom_left")
+
+    elif top_right is None:
+        top_right = SkyCoord(bottom_left.data.lon + width, bottom_left.data.lat + height, frame=bottom_left.frame.name)
```

Comment:
```suggestion
        top_right = SkyCoord(bottom_left.unit_spherical.lon + width,
                             bottom_left.unit_spherical.lat + height,
                             frame=bottom_left.frame)
```

Reply:
I think it'll be `bottom_left.spherical.lon` rather than `bottom_left.unit_spherical.lon`